### PR TITLE
Increase workers for `NetworkPolicy` controller

### PR DESCRIPTION
**Pull request by timuthy**
**How to categorize this PR?**

/area performance
/kind enhancement

**What this PR does / why we need it**:
This PR changes the concurrent sync count for the `NetworkPolicy` controller in GRM from `1` to `20`. Having only one worker is insufficient, esp. in lager seed clusters where many `Service`s exists.

The default count was increased to `5` along the way.

**Special notes for your reviewer**:
I followed the handling of the `ManagedResource` controller for which we use `20` workers as well.

**Release note**:

```other operator
The worker count for the [NetworkPolicy controller](https://github.com/gardener/gardener/blob/master/docs/concepts/resource-manager.md#networkpolicy-controller) in GRM was increased to `20`. This is necessary to create and update `NetworkPolicies` in time, esp. on larger seed clusters.
```

### CODE REVIEW

File: pkg/component/shared/resourcemanager.go [hunk 1]
```diff
@@ -80,12 +80,12 @@ func NewRuntimeGardenerResourceManager(
 	image = &imagevector.Image{Repository: repository, Tag: &tag}
 
 	return resourcemanager.New(c, gardenNamespaceName, secretsManager, resourcemanager.Values{
-		ConcurrentSyncs:                           pointer.Int(20),
-		DefaultSeccompProfileEnabled:              defaultSeccompProfileEnabled,
-		DefaultNotReadyToleration:                 defaultNotReadyToleration,
-		DefaultUnreachableToleration:              defaultUnreachableToleration,
-		EndpointSliceHintsEnabled:                 endpointSliceHintsEnabled,
-		NetworkPolicyAdditionalNamespaceSelectors: additionalNetworkPolicyNamespaceSelectors,
```
Comment (shafeeqes) [inline pkg/component/shared/resourcemanager.go:88]: Why was this line removed?

**Review by ScheererJ — COMMENTED** [review]
Thanks a lot for the change. Very good catch. Looks good already.

/lgtm

**Comment by timuthy** [comment]
Thanks for your review @ScheererJ. I noticed a missing test while re-iterating over the changes and added [142af94](https://github.com/gardener/gardener/pull/8035/commits/142af9422c29d7d14f73122f8485f06454bcd6ef). PTAL.

**Review by ScheererJ — APPROVED** [review]
